src/Searchable.php

```diff
@@ -364,4 +364,32 @@ protected static function usesSoftDelete()
     {
         return in_array(SoftDeletes::class, class_uses_recursive(get_called_class()));
     }
+
+    /**
+     * Was the model searchable before the update happened?
+     *
+     * @return bool
+     */
+    public function wasSearchableBeforeUpdate(): bool
+    {
+        return (clone $this)->forceFill($this->getOriginal())->shouldBeSearchable();
+    }
+
+    /**
+     * Was the model searchable before the deletion happened?
+     *
+     * To be used only when the model gets deleted using `delete`.
+     *
+     * @return bool
+     */
+    public function wasSearchableBeforeDelete(): bool
+    {
+        if (! method_exists($this, 'getDeletedAtColumn')) {
+            return $this->shouldBeSearchable();
```

Comment:
Should it rather be:

```suggestion
            return (clone $this)->forceFill($this->getOriginal())->shouldBeSearchable();
```

Here as well?

Reply:
No no. When deleting a model without soft deletes, nothing really happens with the model, it just gets deleted on the database